src/shared/kilocode/cli-sessions/core/SessionManager.ts

```diff
@@ -404,6 +385,39 @@ export class SessionManager {
 
 			let sessionId = this.sessionPersistenceManager.getSessionForTask(taskId)
 
+			if (sessionId) {
+				if (!this.verifiedSessions.has(sessionId)) {
+					this.logger?.debug("Verifying session existence", "SessionManager", { taskId, sessionId })
+
+					try {
+						const session = await this.sessionClient.get({
+							session_id: sessionId,
+							include_blob_urls: false,
+						})
+
+						if (!session) {
+							this.logger?.info("Session no longer exists, will create new session", "SessionManager", {
+								taskId,
+								sessionId,
+							})
+							sessionId = undefined
+						} else {
+							this.verifiedSessions.add(sessionId)
+							this.logger?.debug("Session verified and cached", "SessionManager", { taskId, sessionId })
+						}
+					} catch (error) {
+						this.logger?.info("Session verification failed, will create new session", "SessionManager", {
+							taskId,
+							sessionId,
+							error: error instanceof Error ? error.message : String(error),
+						})
+						sessionId = undefined
+					}
+				} else {
+					this.logger?.debug("Session already verified (cached)", "SessionManager", { taskId, sessionId })
+				}
```

Comment:
In this pr i see a lot of nested ifs like this; I think it would be better to have something like:

```
 if(!pre_condition_a) {
  tryFixPreconditionA
}
if(!pre_condition_a) return
```

But not a blocker for merging obviously